### PR TITLE
fixing restore to work correctly with prototpyes

### PR DESCRIPTION
**Pull request by mattlgy**

### CODE REVIEW

**Comment by mattlgy** [comment]
Huh, I think Travis is broken. It keeps saying my build is broken...

**Comment by mattlgy** [comment]
But for real:

```
mattlgy:~/workspace/stand-in (prototypes) $ npm test

> stand-in@3.2.0 test /home/ubuntu/workspace/stand-in
> belly-button && lab -t 100 -va code


Clean as a whistle! No issues found!

stand-in
  replace
    ✔ 1) replaces a defined method (14 ms)
    ✔ 2) correctly restores a replaced method (3 ms)
    ✔ 3) uses the correct this context (2 ms)
    ✔ 4) provides a mechanism to use the original method (0 ms)
    ✔ 5) provides the stand-in object as the first parameter to the function (3 ms)
    ✔ 6) supports deep paths for replace (1 ms)
    ✔ 7) supports replacing prototype methods (1 ms)
  prototypes
    ✔ 8) replaces a prototype method (0 ms)
    ✔ 9) correctly restores a replaced prototype method (1 ms)
  assertions
    ✔ 10) throws an error if obj is not defined (1 ms)
    ✔ 11) throws an error if obj is not an object (0 ms)
    ✔ 12) throws an error if obj[function] is undefined (1 ms)
    ✔ 13) throws an error if obj[function] is not a function (0 ms)
    ✔ 14) throws an error if fn is not a function (0 ms)
  duplication
    ✔ 15) throws an error if you try to replace without restoring on the same object (1 ms)
    ✔ 16) allows duplication methods if obj is a new instance (7 ms)
    ✔ 17) allows multiple methods to be replaced on a single instance (1 ms)


17 tests complete
Test duration: 47 ms
Assertions count: 45 (verbosity: 2.65)
No global variable leaks detected
Coverage: 100.00%
```

**Comment by cjihrig** [comment]
The build is failing because the old versions of Node don't like the `const`.

File: test/index.js [hunk 1]
```diff
@@ -132,6 +132,44 @@ describe('stand-in', function () {
     });
   });
 
+  describe('prototypes', function () {
+    it('replaces a prototype method', function (done) {
```
Comment (arb) [inline test/index.js]: These are basically the same as the tests a few lines up. How were they passing before but there is still an issue?

**Comment by mattlgy** [comment]
Ah, I see. I updated the travis file to target the newest unstable of node so it should stop complaining about new features. (I switch the `const`s the `var`s.)

File: test/index.js (hunk 1 above)
Comment (mattlgy) [inline test/index.js]: It was actually the restoring that was failing. I moved my test and got rid of the duplicate one.

**Pull request merged**